gon/linear/contour.py

```diff
@@ -302,6 +289,26 @@ def vertices(self) -> Vertices:
         """
         return list(self._vertices)
 
+    def index(self) -> None:
```

Comment:
I *think* in Shapely there is something similar and is called a "prepared geometry". 
See: [https://shapely.readthedocs.io/en/latest/manual.html#prepared-geometry-operations](https://shapely.readthedocs.io/en/latest/manual.html#prepared-geometry-operations)

Reply:
`prepare` sounds less obvious to me (prepare to what?) than `index` which comes from RDBMS world and describes intention well: we want make queries faster, so indexing may help, it can confused with `Sequence.index` but only the first time, it will not cause any problems since we don't have any plans to support `Sequence` interface for `Contour`s/`Polygon`s

Reply:
we can also call it `preprocess` but it sounds as vague to me as `prepare`